django/contrib/auth/management/commands/createsuperuser.py

```diff
@@ -138,6 +139,14 @@ def handle(self, *args, **options):
                     user_data[PASSWORD_FIELD] = password
             else:
                 # Non-interactive mode.
+                # Load password from environment variable if it is present.
+                if PASSWORD_FIELD in user_data and 'DJANGO_SUPERUSER_PASSWORD' in os.environ:
+                    user_data[PASSWORD_FIELD] = os.environ['DJANGO_SUPERUSER_PASSWORD']
+
+                # Load username from environment variable if it is present unless field is overridden by command
+                username_env_var = 'DJANGO_SUPERUSER_' + self.UserModel.USERNAME_FIELD.upper()
+                if username is None and username_env_var in os.environ:
```

Comment:
You can replace the if condition and branch with this one liner. Set `username` to the value of the environment variable defined by `username_env_var` or to its current value.
```python
username = os.environ.get(username_env_var, username)
```

Reply:
so if I use this, environment variable overrides the user input and this test will fail:
`test_ignore_environment_variable_non_interactive`

Reply:
Yeah, good point.